resources/fetchers/ecr_fetcher.go

```diff
@@ -58,43 +63,44 @@ func (f *ECRFetcher) Stop() {
 
 func (f *ECRFetcher) Fetch(ctx context.Context) ([]fetching.Resource, error) {
 	f.log.Debug("Starting ECRFetcher.Fetch")
-
 	results := make([]fetching.Resource, 0)
-	podsAwsRepositories, err := f.getAwsPodRepositories(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("could not retrieve pod's aws repositories: %w", err)
-	}
-	ecrRepositories, err := f.ecrProvider.DescribeRepositories(ctx, podsAwsRepositories)
+	ecrRepositories := make([]ecr.Repository, 0)
+	podsList, err := f.kubeClient.CoreV1().Pods("").List(ctx, metav1.ListOptions{})
 	if err != nil {
-		return nil, fmt.Errorf("could retrieve ECR repositories: %w", err)
+		logp.Error(fmt.Errorf("failed to get pods  - %w", err))
+		return nil, err
 	}
 
+	for _, podDescriber := range f.PodDescribers {
+		ecrDescribedRepositories, err := f.describeAwsRepositories(ctx, podsList, podDescriber)
+		if err != nil {
+			return nil, fmt.Errorf("could not retrieve pod's aws repositories: %w", err)
+		}
+		ecrRepositories = append(ecrRepositories, ecrDescribedRepositories...)
+	}
 	results = append(results, ECRResource{ecrRepositories})
-	return results, err
+
+	return results, nil
 }
 
-func (f *ECRFetcher) getAwsPodRepositories(ctx context.Context) ([]string, error) {
-	podsList, err := f.kubeClient.CoreV1().Pods("").List(ctx, metav1.ListOptions{})
-	if err != nil {
-		logp.Error(fmt.Errorf("failed to get pods  - %w", err))
-		return nil, err
-	}
+func (f *ECRFetcher) describeAwsRepositories(ctx context.Context, podsList *v1.PodList, describer PodDescriber) ([]ecr.Repository, error) {
```

Comment:
Not a must only if you agree, I would rename it to `describePodImagesRepositories`